src/main/java/echo/TaskList.java

```diff
@@ -0,0 +1,118 @@
+package echo;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * Represents a list of Task objects.
+ * It provides methods to operate on a target Task object in the list.
+ */
+public class TaskList implements Serializable {
+    private ArrayList<Task> allTasks = new ArrayList<>();
```

Comment:
I generally checked all classes and seems like you didnt make any of the attributes public. nice! ✅